File: src/main/java/com/github/zxh/classpy/classfile/constant/ConstantClassInfo.java
package com.github.zxh.classpy.classfile.constant;

/*
CONSTANT_Class_info {
    u1 tag;
    u2 name_index;
}
*/
public class ConstantClassInfo extends ConstantInfo {

    {
        u2("name_index");
    }

    public int getNameIndex() {
        return super.getUInt("name_index");
    }

    @Override
    protected String loadDesc(ConstantPool pool) {
        return pool.getUtf8String(getNameIndex());
    }
    
}

File: src/main/kotlin/com/github/zxh/classpy/classfile/constant/ConstantClassInfo.kt
package com.github.zxh.classpy.classfile.constant;

/*
CONSTANT_Class_info {
    u1 tag;
    u2 name_index;
}
*/
class ConstantClassInfo : ConstantInfo() {

    init {
        u2("name_index")
    }

    fun getNameIndex(): Int {
        return super.getInt("name_index")
    }

    override fun loadDesc(cp: ConstantPool): String {
        return cp.getUtf8String(getNameIndex());
    }
    
}
